### PR TITLE
Race condition in tests

### PR DESCRIPTION
**Pull request by koshatul**
Race condition in tests causes all the current issues with tests failing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 REPO_NAME = go-plugins-helpers
 REPO_OWNER = docker
 PKG_NAME = github.com/${REPO_OWNER}/${REPO_NAME}
-IMAGE = golang:1.7
+IMAGE = golang:1.8
```
Comment (vdemeester) [inline Makefile]: :sweat_smile: 
We may even bump this higher (1.10), but let's do that in a follow-up PR :wink:

**Comment by dave-tucker** [comment]
LGTM!

**Comment by dave-tucker** [comment]
Sorry @cpuguy83 I merged before I saw your comments! The unrelated changes were #105 plus things to appease the linter that I'd attempted in #101, but I never got around to debugging the race in the tests

**Comment by koshatul** [comment]
Thanks for reviewing it.

Sorry, I branched off the changes I made for #105, next time I'll make sure to keep the changes confined to the PR.

**Comment by cpuguy83** [comment]
No worries. Thanks!

**Pull request merged**